library/blob.c

```diff
@@ -491,6 +695,23 @@ static int eblob_check_disk_one(struct eblob_iterate_local *loc)
 		}
 	}
 
+	if ((ctl->flags & EBLOB_ITERATE_FLAGS_VERIFY_CHECKSUM) &&
+	    !(dc->flags & BLOB_DISK_CTL_REMOVE) &&
+	    !(dc->flags & BLOB_DISK_CTL_UNCOMMITTED)) {
+		struct eblob_write_control wc;
+		memset(&wc, 0, sizeof(wc));
+		eblob_rctl_to_wc(&rc, &wc);
+		eblob_dc_to_wc(dc, &wc);
+
+		err = eblob_verify_checksum(bc->back, &dc->key, &wc);
+		if (err) {
+			eblob_dump_wc(bc->back, &dc->key, &wc, "eblob_check_disk_one: checksum verification failed", err);
+			eblob_mark_entry_removed_purge(bc->back, &dc->key, &rc);
```

Comment:
You should 'fix' those keys before run iterator with checksum verification. It removes keys after failed verification to cleanup broken keys. Previously, dnet_recovery makes this by itself, now, when we start to use copy_iterator, it is better to remove them here.